frontend/javascripts/viewer/model/reducers/proofreading_reducer.ts

```diff
@@ -0,0 +1,82 @@
+import update from "immutability-helper";
+import { getVisibleSegmentationLayer } from "viewer/model/accessors/dataset_accessor";
+import type { WebknossosState } from "viewer/store";
+import type { ProofreadAction } from "../actions/proofread_actions";
+
+function ProofreadingReducer(state: WebknossosState, action: ProofreadAction): WebknossosState {
+  const visibleSegmentationLayer = getVisibleSegmentationLayer(state);
+  const layerName = visibleSegmentationLayer?.name;
+  if (!layerName) {
+    return state;
+  }
+  switch (action.type) {
+    case "TOGGLE_SEGMENT_IN_PARTITION": {
+      const minCutPartitions = state.localSegmentationData[layerName].minCutPartitions;
+      const otherPartitionIndex = action.partition === 1 ? 2 : 1;
+      if (minCutPartitions.agglomerateId !== action.agglomerateId) {
+        // TODOM: Discuss whether to instead ignore the action and show an error toast that the new segment does not belong to the same agglomerate.
```

Comment:
TODO :D

What do you think about this?
Currently, if a segment of a different agglomerate is toggled, the action simply clears the current selection which might be unwanted. But maybe we could also develop it this way once requested